### PR TITLE
[WIP] Add test for hex.outdated in umbrella projects

### PR DESCRIPTION
**Pull request by milmazz**
@ericmj I'm trying to add a test for #276 (`hex.outdated` and `hex.outdated --all` in umbrella projects) and I'm getting the following error when I try `hex outdated --all`:

```
  1) test umbrella projects (Mix.Tasks.Hex.OutdatedTest)
     test/mix/tasks/hex/outdated_test.exs:190
     No message matching {:mix_shell, :info, [^ex_doc]} after 0ms.
     The following variables were pinned:
       ex_doc = "\e[1mex_doc\e[0m      0.0.1\e[0m    \e[31m0.1.0\e[0m   \e[31m~> 0.0.1\e[0m     \e[0m"
     Process mailbox:
       {:mix_shell, :info, ["\e[4mDependency\e[0m  \e[4mCurrent\e[0m  \e[4mLatest\e[0m  \e[4mRequirement\e[0m  \e[0m"]}
       {:mix_shell, :info, [""]}
       {:mix_shell, :info, ["A green version in latest means you have the latest version of a given package. A green requirement means your current requirement matches the latest version."]}
```

Any ideas on how can I solve this error?

As I mentioned in #276, doing all the steps manually works fine.

### CODE REVIEW

**Comment by ericmj** [comment]
Does umbrellas work locally after your fix, does it only break in the tests?

**Comment by milmazz** [comment]
> Does umbrellas work locally after your fix, does it only break in the tests?

Yes, exactly! 

It seems that in the unit test phase the `hex.outdated --all` command does not return messages.

**Comment by ericmj** [comment]
Okay, I will investigate. Thanks for the test!

File: test/mix/tasks/hex/outdated_test.exs [hunk 1]
```diff
@@ -186,4 +186,62 @@ defmodule Mix.Tasks.Hex.OutdatedTest do
...
+        bar = [:bright, "bar", :reset, "         ", "0.1.0", :reset, "    ", :green, "0.1.0", :reset, "   ", :green, "0.1.0", :reset, "        "]
+              |> IO.ANSI.format
+              |> List.to_string
+
+        Mix.Task.run "hex.outdated"
+        assert_received {:mix_shell, :info, [^ex_doc]}
+        refute_received {:mix_shell, :info, [^bar]}
+
+        # FIXME: This does not work.
+        # Mix.Task.run "hex.outdated", ["--all"]
```
Comment (ericmj) [inline test/mix/tasks/hex/outdated_test.exs]: The issue is that mix only allows a task to run once. You can do `Mix.Task.reenable "hex.outdated"` or call the task directly with `Mix.Tasks.Hex.Outdated.run`.

Comment (milmazz) [inline test/mix/tasks/hex/outdated_test.exs]: Even if I do the following:

```elixir
        # Mix.Task.run "hex.outdated"
        # assert_received {:mix_shell, :info, [^ex_doc]}
        # refute_received {:mix_shell, :info, [^bar]}

        # Mix.Tasks.Hex.Outdated.run ["--all"]
        Mix.Task.run "hex.outdated", ["--all"]
        assert_received {:mix_shell, :info, [^ex_doc]}
        assert_received {:mix_shell, :info, [^bar]}
```

I got the same result: `no message matching {:mix_shell, :info, [^bar]}`, which represents the dependency for the sub-project `:bacon`

What I'm doing wrong? 😞

Comment (milmazz) [inline test/mix/tasks/hex/outdated_test.exs]: @ericmj The problem was the configuration in the `Mixfile` for the umbrella app, it should work now :)

**Comment by ericmj** [comment]
:heart:

**Pull request merged**